File: sdk/typescript/examples/07-progress.ts
/**
 * Progress Tracking
 */
import { FlashQ, Worker } from '../src';

const client = new FlashQ();

// Worker with progress updates
const worker = new Worker('upload', async (job) => {
  const steps = ['Reading', 'Processing', 'Encoding', 'Uploading', 'Done'];
  for (let i = 0; i < steps.length; i++) {
    await client.progress(job.id, (i + 1) * 20, steps[i]);
    console.log(`Progress: ${(i + 1) * 20}% - ${steps[i]}`);
    await new Promise(r => setTimeout(r, 300));
  }
  return { uploaded: true };
}, { concurrency: 1 });

await worker.start();

// Push job
await client.push('upload', { file: 'video.mp4' });

// Wait for completion
await new Promise(r => setTimeout(r, 3000));

await worker.stop();
await client.close();

File: sdk/python/examples/07_progress.py
"""
Example 07: Progress Tracking

Demonstrates job progress updates.
"""

import asyncio
from flashq import FlashQ


async def main():
    async with FlashQ() as client:
        # Push a job
        job_id = await client.push("long-tasks", {"steps": 10})
        print(f"Pushed job: {job_id}")

        # Pull and process with progress updates
        job = await client.pull("long-tasks")
        if job:
            print(f"Processing job {job.id}...")

            total_steps = job.data.get("steps", 10)
            for step in range(1, total_steps + 1):
                await asyncio.sleep(0.2)

                progress = int((step / total_steps) * 100)
                await client.progress(job.id, progress, f"Step {step}/{total_steps}")
                print(f"  Progress: {progress}%")

            # Get final progress
            progress, message = await client.get_progress(job.id)
            print(f"Final progress: {progress}% - {message}")

            await client.ack(job.id)
            print("Job completed!")


if __name__ == "__main__":
    asyncio.run(main())
